### PR TITLE
security: bump to Java 17 and temurin images

### LINKED ISSUE #180: Upgrade from Java 11?
Any particular reason why PoS apps are with Java 11? Any plan to move them to 17 (LTS) or 18 or even 19?

FYI:
- Online Boutique is now in [18](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/src/adservice), soon in [19](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1132)
- BoA is now in [17](https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/main/src/balancereader/pom.xml#L35):

### PR DESCRIPTION
**Pull request by bourgeoisor**
### Fixes #180, #181

This PR bumps Java from 11 -> 17, and starts using Temurin images.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -142,7 +142,7 @@
           <version>${jib-maven.plugin.version}</version>
           <configuration>
             <from>
-              <image>gcr.io/distroless/java11-debian11</image>
+              <image>eclipse-temurin:17.0.4.1_1-jre-alpine</image>
```
Comment (mathieu-benoit) [inline pom.xml:145]: This section needs to be added in the 2 others apps/`pom.xml` files in order to override the default value.
```
<from>
  <image>eclipse-temurin:17.0.4.1_1-jre-alpine</image>
</from>
```

**Review by mathieu-benoit — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (bourgeoisor) [reply pom.xml:145]: Is this not the root POM that gets used for all services?

Comment (mathieu-benoit) [reply pom.xml:145]: Oh ok gotcha, didn't see this root `pom.xml` properly, yep, all good then :)

**Comment by bourgeoisor** [comment]
The errors all look like this:
```<Class> has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0```

**Comment by bourgeoisor** [comment]
Waiting on a new image of Skaffold that supports 17: https://github.com/GoogleContainerTools/skaffold/issues/7076
Should make it for the next release of Skaffold.

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.173.137.124**](http://34.173.137.124)
- **Embedded H2** DB backed deployment: [****](http://)

**Review by NimJay — APPROVED** [review]
Thanks for this update, @bourgeoisor and @Shabirmean.
All checks are passing.

File: docs/local-dev.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ developement or run the services of the application locally as well.
 - [kubectl CLI](https://kubernetes.io/docs/tasks/tools/)
 - [skaffold CLI](https://skaffold.dev/docs/install/)
 - A Google Cloud Project
-- Java **(Version 11)**
+- Java **(Version 17)**
```
Comment (NimJay) [inline docs/local-dev.md:12]: Thanks for updating the docs too!

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.173.137.124**](http://34.173.137.124)
- **Embedded H2** DB backed deployment: [**35.192.18.102**](http://35.192.18.102)

**Pull request merged**